config/api_config.go

```diff
@@ -57,7 +57,11 @@ func (cfg *APIConfig) SetDefaults() {
 // URL provides the API URL for a given endpoint key.
 func (cfg *APIConfig) URL(key string, args ...interface{}) string {
 	pattern := fmt.Sprintf("%s%s", cfg.BaseURL, cfg.Endpoints[key])
-	return fmt.Sprintf(pattern, args...)
+	if args != nil {
```

Comment:
I see args being empty not being the standard use case. What do you think about

```go
...
If args == nil {
  return pattern
}

return fmt.Sprintf(...)
}
```

Basically, trying to keep the normal code path at a minimal indentation.